netbox/dcim/models/racks.py

```diff
@@ -271,6 +396,19 @@ def clean(self):
                     })
 
     def save(self, *args, **kwargs):
+        if (not self.pk) and self.rack_type:
```

Comment:
I think will have to take out the (not self.pk) and just copy as if you edit an existing rack that did not have a rack-type and set a rack-type it doesn't get updated correctly - can see below I edited IDF128 and to have rack type same as track, but the weight is still the old one.
<img width="1122" alt="Racks | NetBox 2024-07-16 14-10-05" src="https://github.com/user-attachments/assets/affe4006-963b-49be-818a-3d4233b10ddc">